Source/Particles/Collision/BinaryCollision/ElasticCollisionPerez.H

```diff
@@ -107,18 +107,44 @@ void ElasticCollisionPerez (
                amrex::max(n1,n2), T_R(-1.0/3.0) );
     lmdD = amrex::max(lmdD, rmin);
 
+#if (defined WARPX_DIM_RZ)
+    T_R * const AMREX_RESTRICT theta1 = soa_1.m_rdata[PIdx::theta];
+    T_R * const AMREX_RESTRICT theta2 = soa_2.m_rdata[PIdx::theta];
+#endif
+
     // call UpdateMomentumPerezElastic()
     {
       int i1 = I1s; int i2 = I2s;
       for (int k = 0; k < amrex::max(NI1,NI2); ++k)
       {
+
+#if (defined WARPX_DIM_RZ)
+//When particles are having the same vr, their velocities would change after collision,
+//because they may have different azimuth angles, but their velocities are not supposed to be changed.
+//To fix this, we rotate one particle to be in parallel with the other particle,
+//such that their velocities won't change after collision if they have the same vr.
+//Then, we rotate that particle back.
+          T_R const theta = theta2[I2[i2]]-theta1[I1[i1]];
+          T_R u1xbuf;
+          u1xbuf = u1x[I1[i1]];
```

Comment:
The compiler has an easier time optimizing this:
```suggestion
          T_R const u1xbuf = u1x[I1[i1]];
```